descriptions/api.github.com/api.github.com.yaml

```diff
@@ -16749,31 +16808,224 @@ paths:
+        for the specified tool in your repository.
+        Use the `confirm_delete_url` URL if you are content to remove all analyses for a tool.
+        When you delete the last analysis in a set the value of `next_analysis_url` and `confirm_delete_url`
+        in the 200 response is `null`.
+
+        As an example of the deletion process,
+        let's imagine that you added a workflow that configured a particular code scanning tool
+        to analyze the code in a repository. This tool has added 15 analyses:
+        10 on the default branch, and another 5 on a topic branch.
+        You therefore have two separate sets of analyses for this tool.
+        You've now decided that you want to remove all of the analyses for the tool.
+        To do this you must make 15 separate deletion requests.
+        To start, you must find the deletable analysis for one of the sets,
+        step through deleting the analyses in that set,
+        and then repeat the process for the second set.
+        The procedure therefore consists of a nested loop:
+
+        **Outer loop**:
+        * List the analyses for the repository, filtered by tool.
+        * Parse this list to find a deletable analysis. If found:
+
+          **Inner loop**:
+          * Delete the identified analysis.
+          * Parse the response for the value of `next_analysis_url` and, if found, use this in the next iteration.
+
+        The above process assumes that you want to remove all trace of the tool's analyses from the GitHub user interface, for the specified repository, and it therefore uses the `next_analysis_url` value. Alternatively, you could use the `confirm_delete_url` value, which would leave the last analysis in each set undeleted to avoid removing a tool's analysis entirely.
+      operationId: code-scanning/delete-analysis
+      tags:
+      - code-scanning
+      externalDocs:
+        description: API method documentation
+        url: https://docs.github.com/rest/reference/code-scanning#delete-a-code-scanning-analysis-from-a-repository
+      parameters:
+      - "$ref": "#/components/parameters/owner"
+      - "$ref": "#/components/parameters/repo"
+      - name: analysis_id
+        in: path
+        description: The ID of the analysis, as returned from the `GET /repos/{owner}/{repo}/code-scanning/analyses`
+          operation.
+        required: true
+        schema:
+          type: integer
+      - name: confirm_delete
+        in: query
+        description: 'Allow deletion if the specified analysis is the last in a set.
+          The parameter can be used without a value as the parameter value is not
+          considered. If you attempt to delete the final analysis in a set without
+          using this parameter you''ll get a 400 response with the message: `Analysis
+          is last of its type and deletion may result in the loss of historical alert
+          data. Please specify confirm_delete.`'
```

Comment:
I fear this part is a footgun

> the parameter value is not considered

I can assure you that SDK users will do things like this

```js
octokit.request("DELETE /repos/{owner}/{repo}/code-scanning/analyses/{analysis_id}", {
  owner: "octocat",
  repo: "hello-world",
  analysis_id: 123,
  confirm_delete: false
})
```

Because that's what the parameter name implies. But the resulting request URL will be

```
https://api.github.com/repos/octocat/hello-world/code-scanning/analyses/123?confirm_delete=false
```

which will have the opposite effect of what the user tries to achieve, if I understand the description correctly.

I think for a destructive operation such as this, the `confirm_delete` parameter should be set explicitly to `true`.

There was a similar ambiguity with the `recursive` query parameter of [`GET /repos/{owner}/{repo}/git/trees/{tree_sha}`](https://docs.github.com/en/rest/reference/git#get-a-tree) in the past, now it only permits one of 4 values: `0`, `1`, `"true"`, and `"false"`. I recommend the same should happen here

@imwiss @xuorig

Reply:
@gr2m Thanks for flagging this.

> I think for a destructive operation such as this, the confirm_delete parameter should be set explicitly to true.

My understanding is that this is what's happening here. Specifically, this part of the description leads me to believe that `confirm_delete` has to be set to `true` for the operation to successfully delete the final analysis:

> If you attempt to delete the final analysis in a set without using this parameter you'll get a 400 response with the message: `Analysis is last of its type and deletion may result in the loss of historical alert data. Please specify confirm_delete.`'

Let me confirm in the code.

Reply:
@gr2m I confirm that `confirm_delete` has to be `true` for the deletion of the final analysis to be successful. Does this address the main concern?

Reply:
In that case I would suggest to rephrase

> The parameter can be used without a value as the parameter value is not considered

to something like

> The parameter must be set to `true`

Reply:
what happens if it is set to `false`? Does it return an error or does it ignore the parameter?

Reply:
> what happens if it is set to `false`? Does it return an error or does it ignore the parameter?

If it's set to `false` and the operation is for deleting the final analysis, then it returns a `400` with the following error message: `Analysis is last of its type and deletion may result in the loss of historical alert data. Please specify confirm_delete.`

If it's set to `false` and the user is not deleting the final analysis, then the parameter gets ignored.

I'll take the action item of updating the docs. Thanks for the feedback 😄